cache/test_cache.py

```diff
@@ -17,12 +17,19 @@ class TestCacheDisabled(functional.FunctionalTest):
     # Disable caching
     cache_mode = 0
 
-    def chain(self, uri='/', cache_allowed=True):
+    def chain(self, uri='/', cache_allowed=True, wo_data=False):
         if self.cache_mode == 0:
             cache_allowed = False
         if cache_allowed:
-            return chains.cache_repeated(self.messages, uri=uri)
-        return chains.proxy_repeated(self.messages, uri=uri)
+            test_chains = chains.cache_repeated(self.messages, uri=uri)
+        else:
+            test_chains = chains.proxy_repeated(self.messages, uri=uri)
+        if wo_data:
+            for chain in test_chains:
+                if chain.server_response.status:
```

Comment:
```suggestion
                if chain.server_response:
```

`chain.server_response` can be `None`, but status is always valid.

Reply:
No

Reply:
But need make that

Reply:
I mean if some messages are not configured in the `chains`, they're set to `None`, so comparison to `None` should be correct check there.

Reply:
There is an incorrect `Request` and it was just spinning in my head to do something about it